gesturelibrary/src/double.c

```diff
@@ -41,42 +60,117 @@ int set_on_double_tap(void (*listener)(const double_tap_t*)) {
     }
 }
 
-static void update_double_taps(double_tap_t* double_tap, const stroke_t* stroke, const tap_t* tap) {
-    double_tap->x = stroke->x;
-    double_tap->y = stroke->y;
-    double_tap->t = stroke->t;
+static void update_double_taps(const tap_t* tap, const touch_event_t* event) {
+    int null_index     = -1;  // last seen null or failed double_d spot
+    int complete_index = -1;  // last seen complete spot
+    int update         = 1;
+    int group_index    = -1;  // double tap with the closest time the completed tap, will group with this tap
+    // int group_number;
+
+    // float x_diff;
+    // float y_diff;
+    float pos_diff;
+    float time_diff;
+
+    // printf("printing tap_state: %d\n", tap->state);
+
+    if (tap->state == RECOGNIZER_STATE_COMPLETED) {
+        // printf("entered with tap x = %f\n", tap->x);
+        // if completed, compare it to all double_taps in double_d
+        // look for any possible ones and compare time and position
+
+        for (int i = 0; i < MAX_TOUCHES; i++) {
```

Comment:
Is it possible for one tap event to complete multiple double tap events?
For example, two taps complete simultaneously, and then a third tap completes between the two earlier taps in close enough proximity to both.